compiler.pri

```diff
@@ -227,14 +227,33 @@ unix {
 		QMAKE_OBJECTIVE_CXXFLAGS *= -O3 -march=native -ffast-math -ftree-vectorize -fprofile-use
 	}
 
-	CONFIG(c++11) {
-		# Qt 5 will pass the expected compiler flags
-		# needed for C++11 mode when CONFIG includes c++11.
-		# But Qt 4 won't, so we add it manually.
-		lessThan(QT_MAJOR_VERSION, 5) {
-			QMAKE_CXXFLAGS *= -std=c++11
+	# Allow CONFIG(c++11) to enable C++11 mode
+	# when building against Qt 4.
+	lessThan(QT_MAJOR_VERSION, 5):CONFIG(c++11) {
+		QMAKE_CXXFLAGS *= -std=c++11
+	}
+
+	# Detect Qt >= 5.7. Qt 5.7 and above
+	# require C++11 at the very least.
+	greaterThan(QT_MAJOR_VERSION, 5) {
```

Comment:
The comment confused me, led me into the wrong thinking path.

I see now that the comment is for the next two blocks. Easier to see in code-view rather than this diff.